Repair Tesseract issue

Error occuring when trying to build the project:

CMake Error at CMakeLists.txt:35 (find_package):
  By not providing "FindTesseract.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "Tesseract", but CMake did not find one.

  Could not find a package configuration file provided by "Tesseract" with
  any of the following names:

    TesseractConfig.cmake
    tesseract-config.cmake

  Add the installation prefix of "Tesseract" to CMAKE_PREFIX_PATH or set
  "Tesseract_DIR" to a directory containing one of the above files.  If
  "Tesseract" provides a separate development package or SDK, be sure it has
  been installed.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,16 +9,18 @@ include(CTest)
 include(CheckCXXCompilerFlag)
 include(FetchContent)
 
+list(APPEND CMAKE_PREFIX_PATH tesseract)
+
 # Enable flag if supported
 function(enable_cxx_compiler_flag_if_supported flag)
-    string(FIND "${CMAKE_CXX_FLAGS}" "${flag}" flag_already_set)
-    if(flag_already_set EQUAL -1)
-        check_cxx_compiler_flag("${flag}" flag_supported)
-        if(flag_supported)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag}" PARENT_SCOPE)
-        endif()
-        unset(flag_supported CACHE)
+  string(FIND "${CMAKE_CXX_FLAGS}" "${flag}" flag_already_set)
+  if(flag_already_set EQUAL -1)
+    check_cxx_compiler_flag("${flag}" flag_supported)
+    if(flag_supported)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag}" PARENT_SCOPE)
     endif()
+    unset(flag_supported CACHE)
+  endif()
 endfunction()
 
 # Set C++ standard
@@ -26,9 +28,9 @@ set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 # Gather source and header files
-file (GLOB_RECURSE SOURCES CONFIGURE_DEPENDS "src/*.cpp")
-file (GLOB_RECURSE HEADERS CONFIGURE_DEPENDS "src/*.hpp")
-file (GLOB_RECURSE TESTS CONFIGURE_DEPENDS "tests/*.cpp")
+file(GLOB_RECURSE SOURCES CONFIGURE_DEPENDS "src/*.cpp")
+file(GLOB_RECURSE HEADERS CONFIGURE_DEPENDS "src/*.hpp")
+file(GLOB_RECURSE TESTS CONFIGURE_DEPENDS "tests/*.cpp")
 
 # Check if OpenCV is installed
 find_package(OpenCV REQUIRED)
@@ -67,22 +69,24 @@ add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS})
 
 # Add include directories
 target_include_directories(${PROJECT_NAME}
-	PRIVATE src
-	PRIVATE ${OpenCV_INCLUDE_DIRS}
-    PRIVATE spdlog::spdlog
-    PRIVATE ${Tesseract_INCLUDE_DIRS}
-    PRIVATE ${SYMENGINE_INCLUDE_DIRS}
-    PRIVATE nlohmann_json::nlohmann_json
+  PRIVATE
+    src
+    ${OpenCV_INCLUDE_DIRS}
+    spdlog::spdlog
+    ${Tesseract_INCLUDE_DIRS}
+    ${SYMENGINE_INCLUDE_DIRS}
+    nlohmann_json::nlohmann_json
 )
 
 # Add libraries to link
 target_link_libraries(${PROJECT_NAME}
-	PRIVATE ${OpenCV_LIBS}
-    PRIVATE spdlog::spdlog
-    PRIVATE ${Tesseract_LIBRARIES}
-    PRIVATE ${SYMENGINE_LIBRARIES}
-    PRIVATE tensorflow-lite
-    PRIVATE nlohmann_json::nlohmann_json
+  PRIVATE
+    ${OpenCV_LIBS}
+    spdlog::spdlog
+    ${Tesseract_LIBRARIES}
+    ${SYMENGINE_LIBRARIES}
+    tensorflow-lite
+    nlohmann_json::nlohmann_json
 )
 
 
@@ -91,22 +95,24 @@ add_library(${PROJECT_NAME}_lib STATIC ${SOURCES} ${HEADERS})
 
 # Add include directories
 target_include_directories(${PROJECT_NAME}_lib
-	PRIVATE src
-	PRIVATE ${OpenCV_INCLUDE_DIRS}
-    PRIVATE spdlog::spdlog
-    PRIVATE ${Tesseract_INCLUDE_DIRS}
-    PRIVATE ${SYMENGINE_INCLUDE_DIRS}
-    PRIVATE nlohmann_json::nlohmann_json
+  PRIVATE
+    src
+    ${OpenCV_INCLUDE_DIRS}
+    spdlog::spdlog
+    ${Tesseract_INCLUDE_DIRS}
+    ${SYMENGINE_INCLUDE_DIRS}
+    nlohmann_json::nlohmann_json
 )
 
 # Add libraries to link
 target_link_libraries(${PROJECT_NAME}_lib
-	PRIVATE ${OpenCV_LIBS}
-    PRIVATE spdlog::spdlog
-    PRIVATE ${Tesseract_LIBRARIES}
-    PRIVATE ${SYMENGINE_LIBRARIES}
-    PRIVATE tensorflow-lite
-    PRIVATE nlohmann_json::nlohmann_json
+  PRIVATE
+    ${OpenCV_LIBS}
+    spdlog::spdlog
+    ${Tesseract_LIBRARIES}
+    ${SYMENGINE_LIBRARIES}
+    tensorflow-lite
+    nlohmann_json::nlohmann_json
 )
 
 
@@ -115,9 +121,10 @@ add_executable(tests ${TESTS})
 
 # Link GoogleTest and MathBoard library
 target_link_libraries(tests
-	PRIVATE GTest::gtest_main
-    PRIVATE ${PROJECT_NAME}_lib
-    PRIVATE ${OpenCV_LIBS}
+  PRIVATE
+    GTest::gtest_main
+    ${PROJECT_NAME}_lib
+    ${OpenCV_LIBS}
 )
 
 include(GoogleTest)

--- a/tesseract/LeptonicaConfig.cmake
+++ b/tesseract/LeptonicaConfig.cmake
@@ -1,0 +1,15 @@
+# ===================================================================================
+#  The Tesseract CMake configuration file
+#
+#             ** File generated automatically, do not modify **
+#
+#  Usage from an external project:
+#
+#    This file will define the following variables:
+#      - Leptonica_LIBRARIES             : The list of all imported targets.
+#      - Leptonica_INCLUDE_DIRS          : The Tesseract include directories.
+#
+# ===================================================================================
+
+set(Leptonica_INCLUDE_DIRS "/usr/include/leptonica")
+set(Leptonica_LIBRARIES "/usr/lib/libleptonica.so")

--- a/tesseract/TesseractConfig.cmake
+++ b/tesseract/TesseractConfig.cmake
@@ -1,0 +1,19 @@
+# ===================================================================================
+#  The Tesseract CMake configuration file
+#
+#             ** File generated automatically, do not modify **
+#
+#  Usage from an external project:
+#    In your CMakeLists.txt, add these lines:
+#
+#    This file will define the following variables:
+#      - Tesseract_LIBRARIES             : The list of all imported targets.
+#      - Tesseract_INCLUDE_DIRS          : The Tesseract include directories.
+#
+# ===================================================================================
+
+include(CMakeFindDependencyMacro)
+find_dependency(Leptonica)
+
+set(Tesseract_INCLUDE_DIRS "/usr/include/tesseract")
+set(Tesseract_LIBRARIES "/usr/lib/libtesseract.so")

